.github/workflows/main.yml

```diff
@@ -0,0 +1,31 @@
+name: CI
+on:
+  push:
+    branches:
+      - 'main'
+      - 'versions/*'
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  npm-test:
+    uses: ./.github/workflows/npm-test.yml
+
+  docker:
+    needs: npm-test
+    if: startsWith(github.ref, 'refs/tags/v') || (github.ref == 'refs/heads/main') || startsWith(github.ref, 'refs/heads/versions/')
```

Comment:
Is there a specific reason why these conditions are here? The `docker.yml` file already has the correct conditions right? And if it doesn't I think it makes more sense to add them there. Similar for the other conditions below. That way every CI file is completely responsible for their actions.

Reply:
Docker file has no conditions except for `workflow_call`, else it would trigger separately. We have to set them here. I would prefer to have it in their own files but sadly that's not how it works with github workflows.

Reply:
I meant that internally it still has checks such as `if: startsWith(github.ref, 'refs/tags/v') || (github.ref == 'refs/heads/main')`, but I guess those don't have an impact on whether this part of the CI runs or not, which is what is needed and can't be done otherwise.

Reply:
Those conditionals are more to limit the execution, cause the triggering events of the main CI are quite broad. Issue also is that tags are push events and you cant trigger on those separately.

The conditionals inside the docker file are only to generate the correct metadata, cause that metadata step has different config depending on which case is pushed.